api/app/Models/Pool.php

```diff
@@ -632,43 +636,45 @@ public function scopeOrderByPoolBookmarks(Builder $query, ?array $args): Builder
         return $query;
     }
 
-    public function scopeAuthorizedToView(Builder $query)
+    public function scopeAuthorizedToView(Builder $query): void
     {
-
         /** @var \App\Models\User */
         $user = Auth::user();
 
-        if (! $user) {
-            return $query->where('published_at', '<=', Carbon::now()->toDateTimeString());
+        // can view any pool - return query with no filters added
+        if ($user?->isAbleTo('view-any-pool')) {
+            return;
         }
 
-        if (! $user->isAbleTo('view-any-pool')) {
-            $query->where(function (Builder $query) use ($user) {
-
-                if ($user->isAbleTo('view-team-draftPool')) {
-                    // Only add teams the user can view pools in to the query for `whereHas`
-                    $teams = $user->rolesTeams()->get();
-                    $teamIds = [];
-                    foreach ($teams as $team) {
-                        if ($user->isAbleTo('view-team-draftPool', $team)) {
-                            $teamIds[] = $team->id;
-                        }
+        // we might want to add some filters for some pools
+        $filterCountBefore = count($query->getQuery()->wheres);
+        $query->where(function (Builder $query) use ($user) {
+            if ($user?->isAbleTo('view-team-draftPool')) {
+                // Only add teams the user can view pools in to the query for `whereHas`
+                $teams = $user->rolesTeams()->get();
+                $teamIds = [];
+                foreach ($teams as $team) {
+                    if ($user->isAbleTo('view-team-draftPool', $team)) {
+                        $teamIds[] = $team->id;
                     }
-
-                    $query->orWhereHas('legacyTeam', function (Builder $query) use ($teamIds) {
-                        return $query->whereIn('id', $teamIds);
-                    });
                 }
 
-                if ($user->isAbleTo('view-any-publishedPool')) {
-                    $query->orWhere('published_at', '<=', Carbon::now()->toDateTimeString());
-                }
+                $query->orWhereHas('legacyTeam', function (Builder $query) use ($teamIds) {
+                    return $query->whereIn('id', $teamIds);
+                });
+            }
 
-                return $query;
-            });
+            if ($user?->isAbleTo('view-any-publishedPool')) {
+                $query->orWhere('published_at', '<=', Carbon::now()->toDateTimeString());
+            }
+        });
+        $filterCountAfter = count($query->getQuery()->wheres);
+        if ($filterCountAfter > $filterCountBefore) {
+            return;
         }
 
-        return $query;
+        // fall through - anyone can view a published pool
+        $query->where('published_at', '<=', Carbon::now()->toDateTimeString());
```

Comment:
This is true, but it's already capturedby `view-any-publishedPool`, which is assigned to the baseUser role.

Reply:
Even guests have `view-any-publishedPool`.  But how can you call `isAbleTo` on a null user object?

Reply:
Nope, that's a great point! This is necessary to accommodate guests.